pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2017,16 +2010,44 @@ func buildTracingConfig(config *meshconfig.ProxyConfig, proxyType model.NodeType
 				}
 		}
 
-		// custom tags should only be used for sidecar proxies and should not include
-		// gateways due to client requests from outside of the mesh
-		if len(config.Tracing.CustomTags) != 0 && proxyType == model.SidecarProxy {
```

Comment:
cc @howardjohn @nrjpoddar 

Not trying to sneak changes in without review, but Neeraj and I talked about adding support for custom_tags in Gateways. While there is concern about rouge clients, for people using this feature we just need to document in https://github.com/istio/istio.io/pull/7122 that it should not be used.

Reply:
Here's the broader reasoning:

Currently, if you have tracing turned on at Gateways, rogue clients can abuse host/user agent/path or any other headers to store a large amount of data. Custom tags are no different. Tracing is different than metrics in terms of storage and cardinality.

The only safe way for customers to prevent tracing data client abuse in Gateways is to just disable it or live with maximum header size enforced by Envoys.